### PR TITLE
feat(subagent): expose child Pi session ID in results

### PR DESCRIPTION
**Pull request by stvhay**
## Summary

Archimedes results currently omit the logical session ID of the child Pi process. This PR exposes that existing ID as optional `SubagentResult.childSessionId`. The subagent stream captures Pi's initial JSON `session.id`, returns it without modification, and documents the new result metadata as a vendor-neutral external correlation key.

## Motivation

Each subagent task launches its own fresh `pi --mode json --no-session -p <task>` process. `--no-session` disables transcript persistence, but Pi still creates an in-memory logical session UUID and emits it in the initial JSON `session` event.

Before this PR, Archimedes ignored that event and its package documentation did not claim session or trace correlation support. Callers therefore had no deterministic key for joining a result to external telemetry grouped by the child Pi session.

## Compatibility

- `--no-session` remains unchanged; no child transcripts are persisted.
- `childSessionId` is additive and optional; it is absent if the process exits before a valid session event.
- Spawn arguments and environment, Windows behavior, IPC, cancellation, timeouts, progress payloads, usage accounting, and parallel result ordering are unchanged.
- No extra invocation ID, parent-session transport, distributed trace context, or vendor trace ID is added.

## Testing

CI-equivalent checks:

- `corepack pnpm install --frozen-lockfile` — passed with pinned pnpm 10
- `corepack pnpm -r exec -- tsc --noEmit` — passed across all 9 workspace packages
- `corepack pnpm test` — 223 passed

Additional focused checks:

- `corepack pnpm --dir packages/subagent exec tsc --noEmit` — passed
- `corepack pnpm --dir packages/subagent exec vitest run src/stream.test.ts` — 2 passed
- manual one-shot run — returned `childSessionId` exactly matched the child Pi session and enabled exact lookup of its single child trace

The completed design record is under `docs/plans/done/2026-08-05-subagent-session-id.md`, and `docs/plans/README.md` records it as completed. Package release remains a separate maintainer action.

Companion: [pi-langfuse #9](https://github.com/gooyoung/pi-langfuse/pull/9) uses Pi's public logical session ID for ephemeral roots.

## Development disclosure

This PR was developed with assistance from AI tooling. I reviewed the changes in full and take responsibility for their correctness.

### CODE REVIEW

File: docs/plans/done/2026-08-05-subagent-session-id.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+# Expose subagent Pi session IDs
```
[the remark this answers is not in the document]
Comment (stvhay) [reply docs/plans/done/2026-08-05-subagent-session-id.md:1]: I put it in done/ because that seems to be where completed plans are.

**Review by danielcherubini — APPROVED** [review]
### 🎉 Praise

- **Minimal, well-scoped change** — touches only the subagent package's stream/types/README + plan docs. No ripple effects on other packages.
- **Good test coverage** — `stream.test.ts` covers both the happy path (valid session ID captured) and edge cases (missing `id`, non-string `id`). The fake child process pattern is clean and reusable.
- **Defensive capture** — `typeof event.id === "string" && event.id` correctly guards against `null`, numbers, empty strings, etc.
- **Clean optional field** — conditional spread keeps the field off the result entirely when absent (better than `undefined` for JSON consumers).
- **Complete plan doc** — thorough with compatibility boundaries, acceptance criteria, and verification steps.
- **Comment updated** — the ignore list comment correctly removes `session` from the ignored events.

### 🟡 Nit

- **Plan file naming convention** — All existing plans in \`docs/plans/done/\` use the \`plan-NNN-name.md\` format. This is the first to use date-based naming. Minor inconsistency, not blocking.

### 💡 Suggestion

- Consider adding a test for \`id: ""\` (empty string) — the current guard handles it, but an explicit test would make that intent clearer.

**Verdict: Approve** — Clean, minimal, well-tested. Safe to merge.
